hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -2037,11 +2037,11 @@ private void listStatusFindKeyInTableCache(
                 // if entry is a directory
                 if (!deletedKeySet.contains(entryInDb)) {
                   if (!entryKeyName.equals(immediateChild)) {
-                    OmKeyInfo fakeDirEntry = new OmKeyInfo.Builder()
-                        .setVolumeName(omKeyInfo.getVolumeName())
-                        .setBucketName(omKeyInfo.getBucketName())
-                        .setKeyName(immediateChild)
-                        .build();
+                    OmKeyInfo fakeDirEntry = createDirectoryKey(
```

Comment:
It looks to me an opportunity to add a unit test, can we? Thanks!

Reply:
Sure.

Reply:
I find an easy way to reproduce the issue.  Use "key put" to upload an object with a path like object name, for example "object-dir/object-name", then use "fs -ls" to list the bucket. 
Will try to add a UT too.